src/gl/texture.c

```diff
@@ -1690,10 +1690,10 @@ void gl4es_glTexSubImage2D(GLenum target, GLint level, GLint xoffset, GLint yoff
                 free(ndata);
         }
         // check if max_level is set... and calculate higher level mipmap
-        int genmipmap = 0;
+        uint8_t genmipmap = 0;
         if(((bound->max_level == level) && (level || bound->mipmap_need)))
             genmipmap = 1;
-        if(callgeneratemipmap && (level==0) || (level==bound->max_level))
+        if((callgeneratemipmap && (level==0)) || (level==bound->max_level))
```

Comment:
My bad, I should have put the `()` there.
The intent was 
```
if(callgeneratemipmap && ((level==0) || (level==bound->max_level))
```
IIRC.

Reply:
Oh, so that's a but right there! The C standard will do first the && operation and then the || operation! So I just put the braces right there assuming you were relying on operator precedence. I can change it to the other side if that's what you were intending to do.

Reply:
I've pushed the corrected version now.

Reply:
Yep, that was a bug indeed!